proposals/001-ClusterRepresentation.md

```diff
@@ -0,0 +1,305 @@
+<!-- This template is provided as an example with sections you may wish to comment on with respect to your proposal. Add or remove sections as required to best articulate the proposal. -->
+
+# Cluster Representation in the Model
+
+This proposal introduces the concept of _cluster_ into the model.
+
+A _downstream cluster_ is a kafka cluster that the clients connect to.   The downstream cluster can be thought of as a 'virtual' cluster or a 'facade' to
```

Comment:
Yeah. I'm much less sure about `physical` too. Toyed with `underlying` and `target` but `physical` worked well for as being the opposite of `virtual` and really drives the point home for me. 

As far as each Kroxy is concerned each upstream cluster is the `physical` one even if it happens to be another kroxy instance thats an implementation detail that doesn't concern the given instance (otherwise we have a leaky abstraction - but I can see arguments for co-operation between proxies...).

Reply:
I struggled with the `upstream` and `downstream` terms as well, I think `physical` and `virtual` are much better replacements and are explicit in nature to spare the users any non-intuitive mental model mapping (e.g. "oh it mentions "cluster" only, so this must be the physical one, since the other one is called "virtual" ) 

I think either `virtual` and `physical` or `virtual` and `target` would be good alternatives

Reply:
I'll adopt virtual/physical nomenclature.

Reply:
Done